packages/react-dom/src/events/DOMModernPluginEventSystem.js

```diff
@@ -667,32 +716,35 @@ export function accumulateTwoPhaseListeners(
             const listener = listenersArr[i];
             const {callback, capture, type} = listener;
             if (type === targetType) {
-              if (capture === true) {
-                capturePhase.push(
+              if (capture && inCapturePhase) {
```

Comment:
I'm really struggling to follow what's happening here due to very similar naming. E.g. `capture`, `captured`, and `inCapturePhase` probably represent something different but I'm getting lost which is which because they're all used in the same function.